qa/workunits/rbd/rbd_mirror.sh

```diff
@@ -29,9 +29,30 @@ fi
 . $(dirname $0)/rbd_mirror_helpers.sh
 setup
 
-testlog "TEST: add image and test replay"
+testlog " - mirror default namespace with remote-namespace"
 start_mirrors ${CLUSTER1}
 image=test
+NS0=ns0
+rbd --cluster ${CLUSTER1} mirror pool enable ${POOL}
+rbd --cluster ${CLUSTER2} mirror pool enable ${POOL}
+rbd --cluster ${CLUSTER1} namespace create ${POOL}/${NS0}
+rbd --cluster ${CLUSTER2} mirror pool enable ${POOL} ${MIRROR_POOL_MODE} --remote-namespace ${NS0}
+rbd --cluster ${CLUSTER1} mirror pool enable ${POOL}/${NS0} ${MIRROR_POOL_MODE} --remote-namespace ""
+create_image_and_enable_mirror ${CLUSTER2} ${POOL} ${image} ${RBD_MIRROR_MODE}
+wait_for_image_replay_started ${CLUSTER1} ${POOL}/${NS0} ${image}
+write_image ${CLUSTER2} ${POOL} ${image} 100
+wait_for_replay_complete ${CLUSTER1} ${CLUSTER2} ${POOL}/${NS0} ${POOL} ${image}
+wait_for_replaying_status_in_pool_dir ${CLUSTER1} ${POOL}/${NS0} ${image}
+compare_images  ${CLUSTER1} ${CLUSTER2} ${POOL}/${NS0} ${POOL} ${image}
+remove_image_retry ${CLUSTER2} ${POOL} ${image}
+wait_for_image_present ${CLUSTER1} ${POOL}/${NS0} ${image} 'deleted'
+rbd --cluster ${CLUSTER1} mirror pool disable ${POOL}/${NS0}
+rbd --cluster ${CLUSTER1} mirror pool enable ${POOL}
+rbd --cluster ${CLUSTER2} mirror pool enable ${POOL}
+rbd --cluster ${CLUSTER1} mirror pool enable ${POOL} ${MIRROR_POOL_MODE}
+rbd --cluster ${CLUSTER2} mirror pool enable ${POOL} ${MIRROR_POOL_MODE}
+
```

Comment:
can we please add tests for both `default to non-default` and `non-default to default` namespace mirroring?

Reply:
> can we please add tests for both default to non-default and non-default to default namespace mirroring?

Done in `qa/workunits/rbd/rbd_mirror.sh`.

> also can we add some negative test like enabling mirror for the already enabled default namespace it should fail?

Done in `src/test/pybind/test_rbd.py`.